browser/android/background_video/background_video_playback_tab_helper_browsertest.cc

```diff
@@ -0,0 +1,388 @@
+#include "chrome/test/base/chrome_test_utils.h"
+#include "chrome/test/base/testing_browser_process.h"
+#include "content/public/common/content_client.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/browser_test_utils.h"
+#include "content/public/test/content_mock_cert_verifier.h"
+#include "content/public/test/test_navigation_observer.h"
+#include "net/dns/mock_host_resolver.h"
+#include "net/test/embedded_test_server/embedded_test_server.h"
+
+namespace {
+constexpr char kReplaceCallCount[] = "window.getReplaceCallCount()";
+
+}  // namespace
+
+class AndroidBackgroundVideoPlaybackBrowserTest : public PlatformBrowserTest {
+ public:
+  AndroidBackgroundVideoPlaybackBrowserTest()
+      : https_server_(net::EmbeddedTestServer::TYPE_HTTPS) {}
+
+  ~AndroidBackgroundVideoPlaybackBrowserTest() override = default;
+
+  void SetUpCommandLine(base::CommandLine* command_line) override {
+    PlatformBrowserTest::SetUpCommandLine(command_line);
+    mock_cert_verifier_.SetUpCommandLine(command_line);
+  }
+
+  void SetUpInProcessBrowserTestFixture() override {
+    PlatformBrowserTest::SetUpInProcessBrowserTestFixture();
+    mock_cert_verifier_.SetUpInProcessBrowserTestFixture();
+  }
+
+  void SetUpOnMainThread() override {
+    PlatformBrowserTest::SetUpOnMainThread();
+    mock_cert_verifier_.mock_cert_verifier()->set_default_result(net::OK);
+    host_resolver()->AddRule("*", "127.0.0.1");
+
+    base::FilePath test_data_dir = GetTestDataDir();
+
+    https_server_.ServeFilesFromDirectory(test_data_dir);
+    content::SetupCrossSiteRedirector(&https_server_);
+    ASSERT_TRUE(https_server_.Start());
+  }
+
+  base::FilePath GetTestDataDir() {
+    base::ScopedAllowBlockingForTesting allow_blocking;
+    return base::PathService::CheckedGet(brave::DIR_TEST_DATA);
+  }
+
+  void TearDownInProcessBrowserTestFixture() override {
+    mock_cert_verifier_.TearDownInProcessBrowserTestFixture();
+    PlatformBrowserTest::TearDownInProcessBrowserTestFixture();
+  }
+
+  content::WebContents* web_contents() {
+    return chrome_test_utils::GetActiveWebContents(this);
+  }
+
+  void ClickAndWaitForNavigation(const std::u16string& script) {
+    content::TestNavigationObserver observer(web_contents());
+    InjectScript(script);
+    observer.WaitForNavigationFinished();
+  }
+
+  void InjectScript(const std::u16string& script) {
+    content::RenderFrameHost* frame = web_contents()->GetPrimaryMainFrame();
+    frame->ExecuteJavaScriptForTests(script, base::NullCallback(),
+                                     content::ISOLATED_WORLD_ID_GLOBAL);
+  }
+
+ protected:
+  // Must use HTTPS because `youtube.com` is in Chromium's HSTS preload list.
+  net::EmbeddedTestServer https_server_;
+
+ private:
+  content::ContentMockCertVerifier mock_cert_verifier_;
+};
+
+// TESTS
+
+IN_PROC_BROWSER_TEST_F(AndroidBackgroundVideoPlaybackBrowserTest,
+                       TestInjectionMatch) {
```

Comment:
the names of these tests don't really help me understand what you are testing. Something like `TestReplaceExperimentalFlagValues`?

Reply:
Test names improved by commit 👉 [`4ec1f0c` (#28593)](https://github.com/brave/brave-core/pull/28593/commits/4ec1f0c4e58f5985a309bbcc632488c155004ab9)